lib/utility/bulk_queue.rb

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'json'
+
+module Utility
+  class BulkQueue
+    class QueueOverflowError < StandardError; end
+
+    # 500 items or 5MB
+    def initialize(count_threshold = 500, size_threshold = 5 * 1024 * 1024)
```

Comment:
Is `count_threshold` a operations number?

Reply:
Yup, maximum number of operations, will `operation_count_threshold` sound better?

Reply:
Yes, I think it's better